CppCoreGuidelines.md

```diff
@@ -9365,73 +9364,75 @@ If you have a naked `new`, you probably need a naked `delete` somewhere, so you
 
 (Simple) Warn on any explicit use of `new` and `delete`. Suggest using `make_unique` instead.
 
-### <a name="Rr-immediate-alloc"></a>R.12: Immediately give the result of an explicit resource allocation to a manager object
+### <a name="Rr-immediate-alloc"></a>R.12: Instead of manual resource allocation, use factory functions that return RAII objects
```

Comment:
The point of R.12 is exactly about calling C/POSIX/WinAPI/etc "explicit resource allocation" interfaces.

Reply:
That's covered on line https://github.com/isocpp/CppCoreGuidelines/pull/1610/files#diff-feb71ecadc563b52e66838adbd6b8e30R9418

Reply:
That line is a footnote on what is shaping up to be a different rule.

Reply:
...that is not to say it's wrong, but perhaps it needs to be more obvious that it covers the old cases of making calls to explicit APIs as the old example showed.

Reply:
"That line is a footnote on what is shaping up to be a different rule." — Eh? The text I was trying to point to is new in this pull request, and belongs to rule `R.12: Instead of manual resource allocation, use factory functions that return RAII objects`. Here it is:

```
If there is no factory wrapper for your RAII type yet, write one!

    void bad(const char *name)
    {
        FILE *f = fopen(name, "r");  // bad
        AutoFclosingFile myfile(f);
        // ...
    }

    void good(const char *name)
    {
        AutoFclosingFile myfile = my::open_file(name, "r");
        // ...
    }
```

Nobody should ever be calling "explicit resource allocation interfaces" directly in user code (conflating their actual business logic with manual resource management); that would violate rules F.2 and F.3 and perhaps F.1. Resources should be managed by resource-management classes.

Reply:
I'm a fan of collapsing of the rules, the "write a wrapper" is what I always call for.